### PR TITLE
[TensorPipe] Use Descriptor::Tensor::sourceDevice in tensorpipe_agent.

### PR DESCRIPTION
**Pull request by beauby**
Test Plan: CI

Reviewed By: lw

Differential Revision: D27661608

### CODE REVIEW

File: torch/csrc/distributed/rpc/tensorpipe_utils.h [hunk 1]
```diff
@@ -14,6 +14,13 @@
 
 namespace tensorpipe {
 class Message;
+#ifdef TP_NEW_API
+class Allocation;
+class Descriptor;
+#else
+using Allocation = Message;
+using Descriptor = Message;
+#endif // TP_NEW_API
```
Comment (mrshenli) [inline torch/csrc/distributed/rpc/tensorpipe_utils.h:23]: I assume there will be an immediate PR to remove these aliases?

**Review by mrshenli — APPROVED** [review]
LGTM! I don't think there will be any conflicts with https://github.com/pytorch/pytorch/pull/54932, but it might worth checking.

File: torch/csrc/distributed/rpc/tensorpipe_utils.h (hunk 1 above)
Comment (beauby) [reply torch/csrc/distributed/rpc/tensorpipe_utils.h:23]: Right, this preprocessor switch is introduced to allow a clean update in fbcode and GitHub at the same time. The `TP_NEW_API` switch will never actually be enabled in GitHub, and the `#ifdef` will simply disappear in the next PR.

**Comment by mrshenli** [comment]
rocm test failures look irrelevant. 

```
18:14:37   test_Conv1d (__main__.TestNN) ... Traceback (most recent call last):
18:14:38   File "test/run_test.py", line 1101, in <module>
18:14:38     main()
18:14:38   File "test/run_test.py", line 1080, in main
18:14:38     raise RuntimeError(err_message)
18:14:38 RuntimeError: test_nn failed! Received signal: SIGSEGV
```

```
20:19:43   test_autocast_torch_fp16 (__main__.TestCuda) ... Traceback (most recent call last):
20:19:55   File "test/run_test.py", line 1101, in <module>
20:19:55     main()
20:19:55   File "test/run_test.py", line 1080, in main
20:19:55     raise RuntimeError(err_message)
20:19:55 RuntimeError: test_cuda failed! Received signal: SIGSEGV
```

**Pull request closed without merging**